internal/graph/check_test.go

```diff
@@ -478,15 +479,11 @@ type resource
 				ResolutionMetadata: &ResolutionMetadata{Depth: 25},
 			})
 
-			// if the branch producing the cycle is reached first, then an error is returned, otherwise
-			// a result is returned if some other terminal path of evaluation was reached before the cycle
-			if err != nil {
-				require.ErrorIs(t, err, ErrCycleDetected)
-			} else {
-				require.False(t, resp.GetAllowed())
-				require.GreaterOrEqual(t, resp.ResolutionMetadata.DatastoreQueryCount, uint32(1)) // min of 1 (x) if x isn't found and it returns quickly
-				require.LessOrEqual(t, resp.ResolutionMetadata.DatastoreQueryCount, uint32(3))    // max of 3 (x, y, z) before the cycle
-			}
```

Comment:
Removal of non-deterministic test cases demonstrates the power of this small change.